src/pkg/scan/sbom/sbom.go

```diff
@@ -0,0 +1,150 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sbom
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/goharbor/harbor/src/common"
+	"github.com/goharbor/harbor/src/lib/config"
+	scanModel "github.com/goharbor/harbor/src/pkg/scan/dao/scan"
+	sbom "github.com/goharbor/harbor/src/pkg/scan/sbom/model"
+
+	"github.com/goharbor/harbor/src/common/rbac"
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/pkg/permission/types"
+	"github.com/goharbor/harbor/src/pkg/robot/model"
+	"github.com/goharbor/harbor/src/pkg/scan"
+
+	v1 "github.com/goharbor/harbor/src/pkg/scan/rest/v1"
+	"github.com/goharbor/harbor/src/pkg/scan/vuln"
+)
+
+const sbomMimeType = "application/vnd.goharbor.harbor.sbom.v1"
+
+func init() {
+	scan.RegisterScanHanlder(v1.ScanTypeSbom, &scanHandler{GenAccessoryFunc: scan.GenAccessoryArt, RegistryServer: getRegistryServer})
+}
+
+// ScanHandler defines the Handler to generate sbom
+type scanHandler struct {
+	GenAccessoryFunc func(scanRep v1.ScanRequest, sbomContent []byte, labels map[string]string, mediaType string, robot *model.Robot) (string, error)
+	RegistryServer   func(ctx context.Context) string
+}
+
+// RequestProducesMineTypes defines the mine types produced by the scan handler
+func (v *scanHandler) RequestProducesMineTypes() []string {
+	return []string{v1.MimeTypeSBOMReport}
+}
+
+// RequestParameters defines the parameters for scan request
+func (v *scanHandler) RequestParameters() map[string]interface{} {
+	return map[string]interface{}{"sbom_media_types": []string{"application/spdx+json"}}
```

Comment:
should we make `application/spdx+json` a constant variable? same comment for line 64.